v2/ordered-changestream-buffer-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/processing/handler/StreamingHandler.java

```diff
@@ -62,20 +64,49 @@ public void process() {
               + taskContext.getShard().getPort()
               + "/"
               + taskContext.getShard().getDbName();
+      if ("mysql".equals(sourceDbType)) {
+        MySqlDao dao =
+            new DaoFactory(
+                    connectString,
+                    taskContext.getShard().getUserName(),
+                    taskContext.getShard().getPassword())
+                .getMySqlDao(shardId);
 
-      MySqlDao dao =
-          new DaoFactory(
-                  connectString,
-                  taskContext.getShard().getUserName(),
-                  taskContext.getShard().getPassword())
-              .getMySqlDao(shardId);
+        InputRecordProcessor.processRecords(
+            records,
+            taskContext.getSchema(),
+            dao,
+            shardId,
+            taskContext.getSourceDbTimezoneOffset());
+        inputBufferReader.acknowledge();
+        dao.cleanup();
+        LOG.info(
+            "Shard "
+                + shardId
+                + ": Successfully processed batch of "
+                + records.size()
+                + " records.");
+      } else if ("postgresql".equals(sourceDbType)) {
```

Comment:
Please use factory/strategy here. Have a base Dao class and have MySQLDao and PostgreSqlDao derive from it. Have the factory create the appropriate Dao object and pass that to InputRecordProcessor. Similar to what is done for the PubSub and Kafka streaming handler.

Reply:
I have push a commit now to address this. Please review . Thank you for the feedback